src/rust/ide/src/view/project_name.rs

```diff
@@ -0,0 +1,100 @@
+//! This module provides a view for project's name which can be used to edit it.
+
+use crate::prelude::*;
+
+use crate::controller;
+
+use ensogl::display;
+use ensogl::display::world::World;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::shape::text::glyph::font;
+use ensogl::display::shape::text::text_field::TextField;
+use ensogl::display::shape::text::text_field::TextFieldProperties;
+use ensogl::data::color;
+
+use nalgebra::Vector2;
+
+/// The project name's view used for visualizing the project name and renaming it.
+#[derive(Debug,Clone,CloneRef)]
+pub struct ProjectName {
+    logger             : Logger,
+    display_object     : display::object::Instance,
+    text_field         : TextField,
+    project_controller : controller::Project
+}
+
+impl ProjectName {
+    /// Create a new ProjectName view.
+    pub fn new
+    ( logger             : impl AnyLogger
+    , world              : &World
+    , project_controller : &controller::Project
+    , fonts              : &mut font::Registry
+    ) -> Self {
+        let logger                = Logger::sub(&logger,"ProjectName");
+        let display_object        = display::object::Instance::new(&logger);
+        let font                  = fonts.get_or_load_embedded_font("DejaVuSansMono").unwrap();
+        let size                  = Vector2::new(600.0,100.0);
+        let base_color            = color::Rgba::new(1.0, 1.0, 1.0, 0.7);
+        let text_size             = 16.0;
+        let text_field_properties = TextFieldProperties{base_color,font,size,text_size};
+        let text_field            = TextField::new(&world,text_field_properties);
+        text_field.set_content(&project_controller.project_name());
+        let project_controller = project_controller.clone();
+        display_object.add_child(&text_field.display_object());
+        Self {logger,display_object,text_field,project_controller}.initialize()
+    }
+
+    fn initialize(self) -> Self {
+        let project_name = self.clone_ref();
+        self.text_field.set_text_edit_callback(move |change| {
+            // If the text edit callback is called, the TextEdit must be still alive.
+            let field_content = project_name.text_field.get_content();
+            let new_name      = field_content.replace("\n","");
+            if change.inserted == "\n" {
```

Comment:
If we have something to confirm edit, do we have also something to abort edit? 
e.g. if we press Escape or move focus away I'd expect that we should restore the name into the text field

Reply:
This is one of my comments in the issue: https://github.com/enso-org/ide/issues/585. But I'll move the comments to the PR to make them more visible.